### PR TITLE
feat(lambda): add `tracingMode` property to a Function

### PR DESCRIPTION
**Pull request by ren-yamanashi**
### Reason for this change

We can set a `tracingMode` for an Function(Lambda) from cloudformation, but this was not supported in the AWS CDK L2 construct.

### Description of changes

Add `tracingMode` property to FunctionProps and set it in the CfnFunction.

### Description of how you validated changes

Added both unit and integration tests.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-lambda/lib/function.ts [hunk 1]
```diff
@@ -625,6 +652,13 @@ export interface FunctionProps extends FunctionOptions {
    * the handler.
    */
   readonly handler: string;
+
+  /**
+   * The function's [AWS X-Ray](https://docs.aws.amazon.com/lambda/latest/dg/services-xray.html) tracing configuration. To sample and record incoming requests, set Mode to Active
+   *
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-tracingconfig.html#cfn-lambda-function-tracingconfig-mode
+   */
+  readonly tracingConfig?: TracingConfig;
```
Comment (ren-yamanashi) [inline packages/aws-cdk-lib/aws-lambda/lib/function.ts:661]: The `tracingConfig` property of L1 Construct has the following type

```ts
  export interface TracingConfigProperty {
    readonly mode?: string;
  }

public tracingConfig?: cdk.IResolvable | CfnFunction.TracingConfigProperty;
```

However, the type of the `mode` property of this `CfnFunction.TracingConfigProperty` is not strict, so L2 Construct does not use `CfnFunction.TracingConfigProperty`.

Instead, I have a more strict type (a type that uses enum for the `mode` property) and specify it.

**Comment by ren-yamanashi** [comment]
I am getting the following error in [Codecov](https://github.com/aws/aws-cdk/actions/runs/12092399455/job/33722084169?pr=32320)

>   aws-apigateway/lib/domain-name.ts:132:19 - warning JSII5019: The property name "domainName" conflicts with the declaring class "DomainName". This will result in renaming the class to "_DomainName" in C#. Consider renaming "domainName".
> 
>  132   public readonly domainName: string;

But it doesn't seem to have anything to do with this MR. is it a problem on the cdk side?

**Comment by ren-yamanashi** [comment]
```sh
     [2024-12-02T23:56:35.036] [ERROR] jsii/compiler - Type model errors prevented the JSII assembly from being created
      warning JSII6: A "peerDependency" on "constructs" at "^10.0.0" means you should take a "devDependency" on "constructs" at "10.0.0" (found "^10.0.0")
      aws-apigateway/lib/domain-name.ts:132:19 - warning JSII5019: The property name "domainName" conflicts with the declaring class "DomainName". This will result in renaming the class to "_DomainName" in C#. Consider renaming "domainName".
      
      132   public readonly domainName: string;
                            ~~~~~~~~~~
      
        aws-apigateway/lib/domain-name.ts:117:14
          117 export class DomainName extends Resource implements IDomainName {
                           ~~~~~~~~~~
          The declaring class is introduced here
      aws-apigateway/lib/gateway-response.ts:217:19 - warning JSII5019: The property name "responseType" conflicts with the declaring class "ResponseType". This will result in renaming the class to "_ResponseType" in C#. Consider renaming "responseType".
```

I don't think this is an error caused by the code I added, but it doesn't seem to happen on the main branch... 🤔

**Review by mazyu36 — COMMENTED** [review]
Thank you for the contribution.
Just one question.

It seems that tracing config has already been implemented, isn't it?
https://github.com/aws/aws-cdk/blob/9c360734422d52d8b1b299b743ea20a5703c4b06/packages/aws-cdk-lib/aws-lambda/lib/function.ts#L1105

Please let me know if I'm mistaken.

**Comment by ren-yamanashi** [comment]
> Thank you for the contribution. Just one question.
> 
> It seems that tracing config has already been implemented. Am I correct?
> 
> https://github.com/aws/aws-cdk/blob/9c360734422d52d8b1b299b743ea20a5703c4b06/packages/aws-cdk-lib/aws-lambda/lib/function.ts#L1105
> 
> Please let me know if I'm mistaken.

@mazyu36 

I apologize, I overlooked that implementation...   
As you said, it was already implemented, so I would like to close this MR 🙏.

**Comment by mazyu36** [comment]
@ren-yamanashi 
Thanks.
I think it's fine to just close the PR.

**Pull request closed without merging**